components/global/core/io.cellery.observability.api/src/test/java/io/cellery/observability/api/exception/APIInvocationExceptionTestCase.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.cellery.observability.api.exception;
+
+import org.json.JSONObject;
+import org.powermock.reflect.Whitebox;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.Map;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+public class APIInvocationExceptionTestCase {
+
+    @Test
+    public void testMappingWithAPIInvocationException() {
```

Comment:
We can also invoke the service through a method call and assert the response or exception. That way the code of the service also will be covered in the tests. Shall we modify this to test the exception through a service? Or else you can add a new test case as well.

Thia applies to other test classes as well.

Reply:
We have written test cases at different levels for the API and the service level test cases are still being written. This was the approach that was taken so far, and might be better to stick with it.